src/Type/Php/ParseUrlFunctionDynamicReturnTypeExtension.php

```diff
@@ -105,19 +123,43 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 			return $this->componentTypesPairedConstantsForLowercaseString[$componentType->getValue()] ?? new ConstantBooleanType(false);
 		}
 
+		if ($urlType->isUppercaseString()->yes()) {
+			return $this->componentTypesPairedConstantsForUppercaseString[$componentType->getValue()] ?? new ConstantBooleanType(false);
+		}
+
 		return $this->componentTypesPairedConstants[$componentType->getValue()] ?? new ConstantBooleanType(false);
 	}
 
-	private function createAllComponentsReturnType(bool $urlIsLowercase): Type
+	private function createAllComponentsReturnType(bool $urlIsLowercase, bool $urlIsUppercase): Type
 	{
+		if ($urlIsLowercase && $urlIsUppercase) {
+			if (
+				$this->allComponentsTogetherTypeForLowercaseString === null
+				&& $this->allComponentsTogetherTypeForLowercaseString === null
+			) {
+				$returnTypes = [
+					new ConstantBooleanType(false),
+					new NullType(),
+					IntegerRangeType::fromInterval(0, 65535),
+					new IntersectionType([new StringType(), new AccessoryLowercaseStringType(), new AccessoryUppercaseStringType()]),
+					$this->createComponentsArray(true, true),
+				];
+
+				$union = TypeCombinator::union(...$returnTypes);
+				$this->allComponentsTogetherTypeForLowercaseString = $union;
+				$this->allComponentsTogetherTypeForUppercaseString = $union;
```

Comment:
It doesn't work this way ; you'll have to create a 
```
allComponentsTogetherTypeForLowercaseAndUppercaseString
```
or you'll cache a wrong value.

Reply:
Removed in 356e0f810